appshell/appshell_extensions_mac.mm

```diff
@@ -408,23 +430,48 @@ int32 ShowOpenDialog(bool allowMulitpleSelection,
     
     [openPanel setAllowedFileTypes:allowedFileTypes];
     
-    [openPanel beginSheetModalForWindow:[NSApp mainWindow] completionHandler:nil];
-    if ([openPanel runModal] == NSOKButton)
+    // cache the browser details, so has to access
+    // the same inside the completionHandler block.
+    CefBrowserDetails browserDetails;
+    browserDetails.m_browser  = browser;
+    browserDetails.m_response = response;
+    
+    s_panel_to_browser[openPanel] = browserDetails;
+    
+    [openPanel beginSheetModalForWindow:[NSApp mainWindow] completionHandler: ^(NSInteger returnCode)
     {
-        NSArray* urls = [openPanel URLs];
-        for (NSUInteger i = 0; i < [urls count]; i++) {
-            selectedFiles->SetString(i, [[[urls objectAtIndex:i] path] UTF8String]);
+        panel_iterator _browserDetails;
```

Comment:
I'm not sure how the blocks work, can we capture both browser and response parameters?
We can capture variables in Lambda (in C++). Something similar available here?